### ISSUE TITLE
Add prompt when opening a local file

### ISSUE THREAD

**Issue by laurent22** [laurent22/joplin#4203]
Whenever opening a local file (with protocol `file://`) we should display a confirmation prompt. Perhaps it should be a dialog like so:

> Open file SOMEFILE?
>
> * OK
> * Cancel
> * Always open files of type TYPE

**Comment by namanarora00** [comment]
Hey @laurent22, can I work on this?

**Comment by GG-Sheridan** [comment]
Should this issue be labeled as Good First Issue?

We are doing research about how to help newcomers onboarding in an open source project and how to accelerate the issue resolving process.
We have developed a tool to automatically label the Good First Issue (GFI), i.e., the issue which is good for newcomers. 
We first extract 72 features from different dimensions of issue reports, such as readability, semantic meaning, associated code snippets, etc, and experimentally investigate to what extent these features correlate with GFI. Results turn out that a large portion of these features do have a medium to large correlation with GFI. 
We then build a machine learning model based on these features to predict GFI. The experiments on a dataset of 12k issues from Github show that 0.90 AUC can be achieved. 

Our tool shows that this issue has a very large possibility to be a GFI. 
Could you please tell us whether it should be labeled as a GFI?
And if you have any comments or suggestions with this tool, could you please tell us? 
Thank you so much.

**Comment by laurent22** [comment]
No, not a good first issue. Spec is not fully developed and implementing this requires a relatively good knowledge of the codebase.

**Comment by laurent22** [comment]
@nishantwrp, it's when a user adds a link to a file directly in a note. For example `[link](file:///path/to/file.txt)`

However the spec needs work, as there's a lot of details to get right before implementing this. That would start by looking at how other applications handle external links, such as VSCode or PDF viewers, and implement something similar or simpler. If you're interested perhaps you can look into this first.

**Comment by nishantwrp** [comment]
Hi, 

I did a little research on the UX, using two tools i.e. `VSCode` and a pdf viewer that comes along with my OS (Deepin OS) known as `Document Viewer`. And here are my observations.

### VSCode

#### Local Files
It doesn't show any prompt when a link with the offline file is clicked and it opens the file in that vscode window itself in a new tab irrespective of the type of file. It the content of the file is text it simply displays the content otherwise it displays something like the image below.

![vscode-offline](https://user-images.githubusercontent.com/36989112/118376897-551df980-b5e8-11eb-8c38-9b84dd2358ee.jpg)


#### Online Links
It displays a prompt in this case which looks something like

![vscode-online](https://user-images.githubusercontent.com/36989112/118376956-aa5a0b00-b5e8-11eb-8f48-68c11987aa32.jpg)

I like the way the online links are handled here, especially the manage trusted domains feature that allows us to specify some special domains that be opened without any prompt.


### Document Viewer
It doesn't show a prompt both in local and online files.


## My Views
According to me, we should implement the prompt for online links first. Probably just a simple (Proceed, Cancel, Copy, Always allow to open external links) prompt first. The ` Always allow to open external links` option can be added in the settings so it can be altered later.

After that, we can do the same for offline links. But I didn't really get the use case of `Always open files of type TYPE` option, I think we can instead make it  `Always open local files`. This option according to me, would be more needed.

PS: Sorry for the delay :)

**Comment by laurent22** [comment]
Thanks for looking into it but I didn't think you would only check VSCode. It's a programmer tool so what works for this app is not necessarily best for Joplin, which can potentially be used by anyone, including non-technical users. "Configure Trusted Domains" is quite technical and doesn't mean much for many people.

What others non-programmer apps do, like Word, Adobe Reader, Discord, Slack, etc.? You don't have to check exactly the apps I mention, maybe you can think of others too.

**Comment by nishantwrp** [comment]
Hi @mak2002, I am not working on this. Feel free to work on this.
